pyanaconda/anaconda.py

```diff
@@ -337,3 +340,27 @@ def writeXdriver(self, root=None):
         f = open("%s/etc/X11/xorg.conf" %(root,), 'w')
         f.write('Section "Device"\n\tIdentifier "Videocard0"\n\tDriver "%s"\nEndSection\n' % self.xdriver)
         f.close()
+
+    def ensure_running_dbus(self):
+        """Ensure suitable DBus is running. If not, start a new session."""
+        if not launcher.is_dbus_session_running():
+            launcher.start_dbus_session()
+
+        launcher.write_bus_address()
+
+    def run_boss(self):
+        bus_proxy = DBus.get_dbus_proxy()
+        bus_proxy.StartServiceByName(DBUS_BOSS_INSTALLATION_NAME,
+                                     DBUS_FLAG_NONE,
+                                     callback=self._check_boss_started)
+
+    def _check_boss_started(self, returned, error):
```

Comment:
This callback is called when the loop is started (which is too late). Please, call `StartServiceByName` synchronously and fail if the Boss fails to start.

Reply:
Good point, I missed that it could be called later when the event is put in the queue... My fault.